soc/xtensa/esp32/Kconfig.soc

```diff
@@ -7,12 +7,19 @@ config SOC_ESP32
 	select CLOCK_CONTROL
 	select CLOCK_CONTROL_ESP32
 
+if SOC_ESP32
+
 config IDF_TARGET_ESP32
 	bool "ESP32 as target board"
 	default y
-	depends on SOC_ESP32
 
 config ESPTOOLPY_FLASHFREQ_80M
 	bool
 	default y
-	depends on SOC_ESP32
+
+config BT_ESP32_RESERVE_DRAM
```

Comment:
You still have `BT_` as namespace here, which I think should be reserved strictly to the Bluetooth stack configuration options. So I'd call this e.g. `ESP32_BT_RESERVE_RAM` instead

Reply:
Hmm, actually this is kind of consistent with the BT_ESP32 symbol you added to `drivers/bluetooth/hci`, but then this symbol could there as well. However, if it's part of the SoC definitions then it shouldn't have the `BT_` namespace IMO.

Reply:
This is a rather subtle distinction, IMO. @carlescufi @joerchan do you guys have any opinions?

Reply:
Since the new HCI driver doesn't directly depend on this (it's the esp32 linker script which does) I'm kind of inclined to think this should stay here in the SoC directory, and then have an esp32-specific namespace.

Reply:
@jhedberg, it makes sense as this definition is not HCI related. Updated the PR.